### PR TITLE
chore: update vite and rollup

### PR DESCRIPTION
**Pull request by yooouuri**

### CODE REVIEW

**Comment by yooouuri** [comment]
No breaking changes as fat as I can see: https://vitejs.dev/guide/migration.html#general-changes

Tested build and serve (dev), both working.

**Comment by yooouuri** [comment]
Seems like they changed something about using ctrl+c to terminate the dev server, using `close` event on `server.httpServer` to detect a termination.

see https://github.com/electron-vite/vite-plugin-electron/pull/118/commits/734c10f8898c7d2f2b82b8d6e2da71d070b8a6e0

File: src/index.ts [hunk 1]
```diff
@@ -16,6 +16,12 @@ export default function electron(config: Configuration | Configuration[]): Plugi
         server.httpServer?.once('listening', () => {
           bootstrap(config, server)
         })
+        server.httpServer?.once('close', () => {
+          if (process.electronApp) {
+            process.electronApp.removeAllListeners()
+            process.electronApp.kill()
+          }
+        })
```
Comment (caoxiemeihao) [inline src/index.ts:24]: Maybe it's not necessary!
https://github.com/vitejs/vite/blob/060cd66878eb6bee53a55ea3ec0afd724ee9b35c/packages/vite/src/node/shortcuts.ts#L119
image
Eventually, the process will exit after the server is close.

**Review by caoxiemeihao — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (yooouuri) [reply src/index.ts:24]: With Vite 4, using ctrl+c without listening for the close event, the Electron process won’t get killed.

So maybe it’s not required for the quit action… But it is for ctrl+c

Comment (caoxiemeihao) [reply src/index.ts:24]: I used Vite4 in [electron-vite-vue](https://github.com/electron-vite/electron-vite-vue), I tried `Ctrl+C` and Electron exited normally.

Comment (caoxiemeihao) [reply src/index.ts:24]: Hahaha, yes, you are right 😆

**Review by caoxiemeihao — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (yooouuri) [reply src/index.ts:24]: Happy to contribute, thanks for this project.

**Pull request merged**